src/ray/gcs/redis_async_context.h

```diff
@@ -25,20 +25,43 @@ extern "C" {
 struct redisAsyncContext;
 struct redisReply;
 typedef void redisCallbackFn(struct redisAsyncContext *, void *, void *);
+
+#include "hiredis/async.h"
+#include "hiredis/hiredis.h"
 }
 
 namespace ray {
-
 namespace gcs {
 
+template <typename RedisContextType>
+struct RedisContextDeleter {
+  RedisContextDeleter(){};
+
+  void operator()(RedisContextType *context) {}
```

Comment:
[nit] you can add `static_assert(false);` here to avoid accidental instantiation of unwanted typed templates.

Reply:
Another approach is to not use template at all, and use a single deleter struct with overloads:

```
struct RedisContextDeleter {
  RedisContextDeleter(){};
  void operator()(redisContext *context) { redisFree(context); }
  void operator()(redisAsyncContext *context) { redisAsyncFree(context); }
};
```

Reply:
you're right, this is safer